### DISCUSSION TITLE
Renderer for main app

### DISCUSSION THREAD

**Discussion by Nowp** [PolyNoradrenalin/AntSimulator#48] [Ideas]
A renderer will be the bridge between the app and the engine.

Using an entity, it will tell how it should be rendered

@jumsee what do you think ?

**Comment by Narvin-Chana** [comment]
> A renderer will be the bridge between the app and the engine.
> 
> Using an entity, it will tell how it should be rendered
> 
> @jumsee what do you think ?

I agree that there should be a bridge between engine and app in order to feed the renderer the data it needs to draw the entity. 

Imo a renderer should belong to the app buuut it should also not directly extract the information. 

I don't know if we should have a renderer for each entity or a default renderer that goes through the entities list and renders them one by one. The problem is that an entity has to have additionnal info (like texture or color) that is specific to the app. Since we want to detach the app from the engine it would be a bad idea to add that information to the entity itself. 

Any thoughts @Nowp ?

**Reply by Nowp** [reply]
I didn't say that we would add information on the entity but that the renderer would get the Transform object to know where it should be rendered, with what angle etc.
All the render specific attributes would be in the renderer (color, texture, RTX, ...)

**Reply by Narvin-Chana** [reply]
Yeah I agree with you on that, the main question I asked concerned how exactly we should render things, (will a renderer class just go through all existing entities and draw them?)

**Reply by Nowp** [reply]
I think we can have a list of all the renderers to draw and iterate through them but where this list should be I don't really know
